"wult deploy" command fails to execute on SUT having kernel with CONFIG_WERROR enabled

The following error is seen when executing "wult deploy" command on a SUT whose Linux kernel is compiled with CONFIG_WERROR kernel configuration set to true.

```
[root@localhost ~]# stats-collect deploy
Nothing to deploy to the local host.

[root@localhost ~]# wult deploy
Compiling the drivers for kernel '6.6.52-11.0s4c40r5.el8.x86_64'
wult: error: the following command failed with exit code 2:
make -C '/tmp/wult-emr2v9sr/drivers' KSRC='/usr/src/kernels/6.6.52-11.0s4c40r5.el8.x86_64'

make: Entering directory '/tmp/wult-emr2v9sr/drivers'
make -C /usr/src/kernels/6.6.52-11.0s4c40r5.el8.x86_64 M=/tmp/wult-emr2v9sr/drivers W=1 modules
make[1]: Entering directory '/usr/src/kernels/6.6.52-11.0s4c40r5.el8.x86_64'
  CC [M]  /tmp/wult-emr2v9sr/drivers/main.o
  CC [M]  /tmp/wult-emr2v9sr/drivers/uapi.o
make[1]: Leaving directory '/usr/src/kernels/6.6.52-11.0s4c40r5.el8.x86_64'
make: Leaving directory '/tmp/wult-emr2v9sr/drivers'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-22)
  You are using:           gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-23)
/tmp/wult-emr2v9sr/drivers/uapi.c:47:12: error: 'set_bool' defined but not used [-Werror=unused-function]
 static int set_bool(struct wult_info *wi, bool *boolptr, bool val)
            ^~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [scripts/Makefile.build:243: /tmp/wult-emr2v9sr/drivers/uapi.o] Error 1
make[2]: *** [/usr/src/kernels/6.6.52-11.0s4c40r5.el8.x86_64/Makefile:1921: /tmp/wult-emr2v9sr/drivers] Error 2
make[1]: *** [Makefile:234: __sub-make] Error 2
make: *** [Makefile:21: default] Error 2

Consider using driver compilation 'make' options such as 'CC='.
```

--- a/drivers/idle/wult/uapi.c
+++ b/drivers/idle/wult/uapi.c
@@ -41,27 +41,6 @@ static int set_enabled(bool enabled)
 	return err;
 }
 
-/*
- * Set a boolean variable pointed to by 'boolptr' to value 'val'.
- */
-static int set_bool(struct wult_info *wi, bool *boolptr, bool val)
-{
-	int err = 0;
-
-	mutex_lock(&wi->enable_mutex);
-	if (*boolptr == val || !wi->enabled)
-		*boolptr = val;
-	else
-		/*
-		 * The measurements must be disabled in order to toggle the
-		 * interrupt focus mode.
-		 */
-		err = -EBUSY;
-	mutex_unlock(&wi->enable_mutex);
-
-	return err;
-}
-
 static ssize_t dfs_write_bool_file(struct file *file,
 				   const char __user *user_buf,
 				   size_t count, loff_t *ppos)

